jest-setup.ts

```diff
@@ -0,0 +1,77 @@
+declare global {
+  namespace jest {
+    interface Matchers<R> {
+      toHaveQueryParams(expectedParamsKeys: Array<string>): R;
+      toHaveQueryParamsValues(expectedParams: Record<string, string>): R;
+      toHaveOrigin(expectedOrigin: string): R;
+    }
+  }
+}
+
+expect.extend({
+  toHaveQueryParams(received, expectedParamsKeys) {
+    const { params } = breakUrl(received);
+    for (let k of expectedParamsKeys) {
+      if (params[k] === undefined) {
+        return {
+          message: () => `URL query parameter [${k}] should exist, but it doesn't`,
+          pass: false,
+        };
+      }
+    }
+    return {
+      message: () =>
+        `URL [${received}] should not include all of the parameters ${JSON.stringify(expectedParamsKeys)}, but it does`,
```

Comment:
This is a bit confusing, but if I understand correctly this is used when with .not ?

Reply:
Yes, I agree it is a bit confusing - and your understanding is correct (or at least the same as mine ;) ). However this is not something I can do much about...